nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/StatelessNiFi.java

```diff
@@ -69,7 +69,7 @@ public static void main(final String[] args) throws IOException, ClassNotFoundEx
             logger.info("Unpacking {} NARs", narFiles.length);
             final long startUnpack = System.nanoTime();
             for (final File narFile : narFiles) {
```

Comment:
I created a nar class loader to work without unpacking the files, I always wonder if in stateless deployments this wouldn't be better.  Although I did not do the docs, but in stateless, do you need the docs?

Reply:
Interesting. I looked into this briefly a couple months ago and realized that we still needed to unpack nars for NiFi, so I didn't bother looking much further. Not sure if the need was only for docs or not, i don't remember the details. Just remember that I stopped looking into it for some reason or another (it's possible that I just forgot about it??) But yeah it could be worth pursuing.